MPM/mpm.php

```diff
@@ -115,7 +117,7 @@ function mpm_webhook()
 				$order->payment_complete();
 			}
 		}
-		elseif ($payment->isOpen() === FALSE)
+		elseif ($payment->isOpen() === FALSE || $onHold)
```

Comment:
Lennard,

 

Sorry voor de late reactie. Mijn Workstation was eergisteren gecrasht en gedeeltelijk de bestanden verloren :( . Nu weer volleidg online!

 

De OnHold is wel nodig voor het checken of  een betaling een payment nodig heeft. Want needs_payment valideerd geen on-hold betalingen. 

 

Op de andere plek bij payment->isOpen() === false   is het inderdaad zoals je zegt niet nodig. ;)

 

Ik heb een PULL request aangemaakt voor je.

 

Ik hoor het wel.

 

From: Lennard van Gunst [mailto:notifications@github.com] 
Sent: donderdag 16 april 2015 11:20
To: mollie/WooCommerce
Cc: George van Engers
Subject: Re: [WooCommerce] Banktransfer orders are placed on-hold. (#29)

 

In MPM/mpm.php <https://github.com/mollie/WooCommerce/pull/29#discussion_r28494816> :

> @@ -115,7 +117,7 @@ function mpm_webhook()
>                              $order->payment_complete();
>                      }
>              }
> -            elseif ($payment->isOpen() === FALSE)
> +            elseif ($payment->isOpen() === FALSE || $onHold)

Deze $onHold check is volgens mij niet nodig. De order moet alleen geupdate worden als de betaling niet meer OPEN is. Wanneer de webhook nu wordt aangeroepen wanneer de order on-hold is maar de payment nog steeds OPEN zal de order status failed krijgen, wat niet de bedoeling is.

—
Reply to this email directly or view it on GitHub <https://github.com/mollie/WooCommerce/pull/29/files#r28494816> .  <https://github.com/notifications/beacon/ADrkE9CmH4zys9Msxxz8jfaeszI3oYqtks5n_3ZKgaJpZM4EBNvU.gif>